lang/csharp/src/apache/test/Util/LogicalTypeTests.cs

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Globalization;
+using Avro.Util;
+using NUnit.Framework;
+
+namespace Avro.Test
+{
+    [TestFixture]
+    class LogicalTypeTests
+    {
+        [TestCase("1234.56")]
+        [TestCase("-1234.56")]
+        [TestCase("123456789123456789.56")]
+        [TestCase("-123456789123456789.56")]
+        [TestCase("000000000000000001.01")]
+        [TestCase("-000000000000000001.01")]
+        public void TestDecimal(string s)
```

Comment:
@petersilverwood suggested including `Decimal.MinValue` and `Decimal.MaxValue` over in the confluent fork. Also, `0.0` would be a good one.

Reply:
There is a `TestDecimalMinMax` test case directly below this one.